sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -748,6 +762,118 @@ myQueue.submit([&](handler &cgh) {
 });
 ----
 
+== Host Side pipe read/write
+
+If the read/write member functions of a pipe are called from the host side, a `sycl::queue` is added to the parameters. The `memory_order` parameter is also added to the parameters for future work.
+
+[source,c++,Host pipe read write members,linenums]
+----
+template <class Name, class DataT, int32_t MinCapacity = 0,
+          class PropertiesT = decltype(oneapi::experimental::properties{})>
+class pipe {
+  // Blocking
+  static _dataT read(queue &Q, memory_order Order = memory_order::seq_cst);
+  static void write(queue &Q, const _dataT &Data, memory_order Order = memory_order::seq_cst);
+  // Non-blocking
+  static _dataT read(queue &Q, bool &Success, memory_order Order = memory_order::seq_cst);
+  static void write(queue &Q, const _dataT &Data, bool &Success, memory_order Order = memory_order::seq_cst);
+}
+----
+
+== Simple example of host-to-device write&read
+
+[source,c++,First example,linenums]
+----
+using default_pipe_properties = decltype(sycl::ext::oneapi::experimental::properties(sycl::ext::intel::experimental::uses_valid<true>));
+
+// Classes used to name the kernels
+class TestTask;
+class H2DPipeID;
+class D2HPipeID;
+
+using H2DPipe = sycl::ext::intel::experimental::pipe<H2DPipeID, int, 10, default_pipe_properties>;
+using D2HPipe = sycl::ext::intel::experimental::pipe<D2HPipeID, int, 10, default_pipe_properties>;
+
+struct BasicKernel {
+  void operator()() const { 
+    auto a = H2DPipe::read();
+    D2HPipe::write(a+1);
+  }
+};
+
+int main() {
+  queue q(testconfig_selector{});
+  H2DPipe::write(q, 1);  
+
+  q.submit([&](handler &h) {
+    h.single_task<TestTask>(BasicKernel{});
+  });
+  auto b = D2HPipe::read(q);
+  std::cout << b << std::endl; // It should print 2;
+}
+----
+
+== Add new member to the handler class leveraged by the Experimental API
+
+Add the following functions to the `sycl::handler` interface described in Section 4.9.4.3 of
+the SYCL 2020 specification.
+
+Add to Table 130, "Member functions of the handler class".
+
+--
+[options="header"]
+|====
+| Member Function | Description
+a| 
+[source, c++]
+----
+void ext_intel_read_host_pipe
+       (const std::string &Name, void *Ptr,
+                       size_t Size, bool Block=false);
+----
+|  Read from a host pipes given a host address.
+
+   `Name` is the name of the host pipe to be passed into lower level runtime.
+   
+   `Ptr` is the host pointer of the host pipe as identified by address of its const expr member.
+   
+   `Size` is the size of data getting read.
+   
+   `Block` is indicating whether read/write opeartion is blocking. Default is false.
+   
+a| 
+
+[source, c++]
+----
+void ext_intel_write_host_pipe
+       (const std::string &Name, void *Ptr,
+                       size_t Size, bool Block=false);
+----
+|  Write to a host pipes given a host address.
+
+   `Name` is the name of the host pipe to be passed into lower level runtime.
```

Comment:
Last question, otherwise I think I'm okay with these changes: How is the user supposed to use this interface? If the name is from the lower level runtime, I don't see how they could possibly get it.

Reply:
@steffenlarsen , thanks. I don't think the user can use this interface directly as they cannot get the name. Right now it is called by the internal function only.

Reply:
@steffenlarsen , as suggested , I have moved them to the private space and removed them from the doc.